### DISCUSSION TITLE
Trouble interpreting a var into a template literals className

### DISCUSSION THREAD

**Discussion by idn8k** [tailwindlabs/tailwindcss#13911] [Help] [answered]
Hi there,
I'm having an issue where every time I try to style an item using template literals the item is rendered randomly, sometimes correctly, and sometimes not to my page.

I just can't figure out what I am doing wrong...

Any idea?
![Xnip2024-06-28_11-31-50](https://github.com/tailwindlabs/tailwindcss/assets/129896903/be1547a4-e8f1-48ad-86cf-ee3752c65029)

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Reply by idn8k** [reply]
Damn! Thx a lot! (:
